source/crud/crud.rst

```diff
@@ -336,165 +340,209 @@ Read
      * Internal replication use only - driver should not set
      *
-     * The default in servers >= 3.2 is no value: no "oplogReplay" option is sent with
-     * the "find" command.
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     * For servers < 3.2, the OplogReplay wire protocol flag is used and defaults to false.
      *
-     * In servers < 3.2, the OplogReplay OP_QUERY flag defaults to false.
-     *
-     * @see http://docs.mongodb.org/meta-driver/latest/legacy/mongodb-wire-protocol/#op-query
+     * @see https://docs.mongodb.com/manual/reference/command/find/
      */
     oplogReplay: Optional<Boolean>;
 
     /** 
      * Limits the fields to return for all matching documents.
-     * The default is no projection.
      *
-     * @see http://docs.mongodb.org/manual/tutorial/project-fields-from-query-results/
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     *
+     * @see https://docs.mongodb.com/manual/reference/command/find/
      */
     projection: Optional<Document>;
 
-    /**
-     * The number of documents to skip before returning.
+    /** 
+     * If true, returns only the index keys in the resulting documents.
+     *
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
      *
-     * In servers < 3.2, this is a wire protocol parameter that defaults to 0.
+     * @see https://docs.mongodb.com/manual/reference/command/find/
+     */
+    returnKey: Optional<Document>;
+
+    /** 
+     * Determines whether to return the record identifier for each document. If true, adds a field $recordId to the returned documents.
      *
-     * The default in servers >= 3.2 is no skip: no "skip" option is sent with
-     * the "find" command.
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
      *
-     * @see http://docs.mongodb.org/manual/reference/method/cursor.skip/
+     * @see https://docs.mongodb.com/manual/reference/command/find/
      */
-    skip: Optional<Int32>;
+    showRecordId: Optional<Document>;
 
     /**
-     * The order in which to return matching documents. If $orderby also exists
-     * in the modifiers document, the sort field overwrites $orderby.
-     * The default is no sort.
+     * The number of documents to skip before returning.
      *
-     * @see http://docs.mongodb.org/manual/reference/method/cursor.sort/
-     */ 
-    sort: Optional<Document>;
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     * For servers < 3.2, this is a wire protocol parameter that defaults to 0.
+     *
+     * @see https://docs.mongodb.com/manual/reference/command/find/
+     */
+    skip: Optional<Int64>;
 
-    /**
-     * Optionally specifies a collation to use in MongoDB 3.4 and higher.
+    /** 
```

Comment:
Trailing whitespace. This isn't the first instance, so you may want to go through the file and trim any occurrences.